packages/core/src/external/carequality/ihe-gateway-v2/invoke-patient-discovery.ts

```diff
@@ -0,0 +1,39 @@
+import { OutboundPatientDiscoveryReq } from "@metriport/ihe-gateway-sdk";
+import { makeLambdaClient } from "../../aws/lambda";
+import { Config } from "../../../util/config";
+import { processAsyncError } from "../../../util/error/shared";
+
+export type IHEGatewayV2XCPDRequestParams = {
+  patientId: string;
+  cxId: string;
+  pdRequestIHEGatewayV2: OutboundPatientDiscoveryReq;
+};
+const iheGatewayV2OutboundPatientDiscoveryLambdaName = "IHEGatewayV2OutboundPatientDiscoveryLambda";
+
+export async function startPatientDiscoveryIHEGatewayV2({
+  pdRequestIHEGatewayV2,
+  patientId,
+  cxId,
+}: {
+  pdRequestIHEGatewayV2: OutboundPatientDiscoveryReq;
+  patientId: string;
+  cxId: string;
+}): Promise<void> {
+  const lambdaClient = makeLambdaClient(Config.getAWSRegion());
+  const params = { patientId, cxId, pdRequestIHEGatewayV2 };
+  console.log(
+    `Invoking IHEGatewayV2 Outbound Patient Discovery Lambda with params: ${JSON.stringify(params)}`
+  );
+  // intentionally not waiting
+  lambdaClient
+    .invoke({
+      FunctionName: iheGatewayV2OutboundPatientDiscoveryLambdaName,
+      InvocationType: "Event",
+      Payload: JSON.stringify(params),
+    })
+    .promise()
+    .catch(
+      processAsyncError("Failed to invoke lambda to poll outbound patient discovery responses")
+    );
```

Comment:
setting up a factory approach

Reply:
I see you created a Factory, but we have the same code on both the lambda and the "direct" implementation.

We should externalize the lambda code to Core and call it from the "direct" implementation, so we can run the same code on local as we run on the lambdas - otherwise we're testing local something that's different from what we execute on the cloud.

Example: packages/lambdas/src/document-downloader.ts

Reply:
Yeah, both call the same `createSignSendProcessXCPDRequest`, but the lambda has additional logic to call the API. Just trying to have them as close as possible to each other, so when we run on local we're very closely running the same logic as we have on the cloud.

Chatted on a huddle and will let you take it from here. 👍